ui/raidboss/data/06-ew/ultimate/dragonsongs_reprise_ultimate.ts

```diff
@@ -1327,78 +983,388 @@ const triggerSet: TriggerSet<Data> = {
+          },
+        };
+
+        const key = matches.id === '6715' ? 'in' : 'out';
+        const inout = output[key]!();
+        data.diveFromGraceLashGnashKey = key;
+
+        const num = data.diveFromGraceNum[data.me];
+        if (num === undefined) {
+          // Don't print a console error here because this response gets eval'd as part
+          // of the config ui and testing.  We'll get errors elsewhere if needed.
+          // TODO: maybe have a better way to know if we're in the middle of testing?
+          return { intoText: inout };
+        }
+
+        if (data.eyeOfTheTyrantCounter === 1) {
+          if (num === 2) {
+            if (!data.diveFromGraceHasArrow[num])
+              return { alertText: output.circlesDive2!({ inout: inout }) };
+            if (data.diveFromGraceDir[data.me] === 'up')
+              return { alertText: output.upArrowDive2!() };
+            if (data.diveFromGraceDir[data.me] === 'down')
+              return { alertText: output.downArrowDive2!() };
+          } else if (num === 3) {
+            return { alertText: output.inOutAndBait!({ inout: inout }) };
+          }
+        } else if (data.eyeOfTheTyrantCounter === 2) {
+          if (num === 2 || (num === 1 && data.diveFromGracePreviousPosition[data.me] === 'middle'))
+            return { alertText: output.inOutAndBait!({ inout: inout }) };
+        }
+
+        // Otherwise, just tell the remaining people the dodge.
+        return { infoText: inout };
+      },
+    },
+    {
+      id: 'DSR Dive From Grace Dive Collect',
+      // 670E Dark High Jump
+      // 670F Dark Spineshatter Dive
+      // 6710 Dark Elusive Jump
+      // Collect players hit by dive
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: ['670E', '670F', '6710'], source: 'Nidhogg' }),
+      run: (data, matches) => {
+        const posX = parseFloat(matches.targetX);
+        data.diveFromGracePositions[matches.target] = posX;
+      },
+    },
+    {
+      id: 'DSR Dive From Grace Post Stack',
+      // Triggered on first instance of Eye of the Tyrant (6714)
+      type: 'Ability',
+      netRegex: NetRegexes.ability({ id: '6714', source: 'Nidhogg', capture: false }),
+      // Ignore targetIsYou() incase player misses stack
+      suppressSeconds: 1,
+      infoText: (data, _matches, output) => {
+        const num = data.diveFromGraceNum[data.me];
+        if (!num) {
+          console.error(`DFG Tower 1 Reminder: missing number: ${JSON.stringify(data.diveFromGraceNum)}`);
+          return;
+        }
+
+        const inout = output[data.diveFromGraceLashGnashKey]!();
+
+        if (data.eyeOfTheTyrantCounter === 1) {
+          // Tower 1 soaks only based on debuffs.
+          if (num === 3) {
+            if (data.diveFromGraceDir[data.me] === 'circle') {
+              if (data.diveFromGraceHasArrow[3])
+                return output.southTower1!({ inout: inout });
+              return output.circleTowers1!({ inout: inout });
```

Comment:
No way this fits, right?
```suggestion
              return data.diveFromGraceHasArrow[3] ? output.southTower1!({ inout: inout }) : output.circleTowers1!({ inout: inout });
```

Reply:
I mean, it maybe does because the line length gets long.  I think I just avoid that when there's parameters because I find the original easier to read than a really long line.